pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiStageEngineIntegrationTest.java

```diff
@@ -558,7 +558,7 @@ public void testBase64Func()
 
     // non-string identifier used in a filter
     sqlQuery = "SELECT fromUtf8(fromBase64(toBase64(toUtf8(AirlineID)))), AirlineID FROM mytable WHERE "
-        + "fromUtf8(fromBase64(toBase64(toUtf8(AirlineID)))) = AirlineID LIMIT 10";
+        + "fromUtf8(fromBase64(toBase64(toUtf8(AirlineID)))) = CAST(AirlineID as VARCHAR) LIMIT 10";
```

Comment:
We allow implicit casting within UDF (same as v1) here, so it might cause surprise when the left hand doesn't require type casting and right hand requires.